lib/fizzy/parser_expr.cpp

```diff
@@ -0,0 +1,31 @@
+#include "parser.hpp"
+
+namespace fizzy
+{
+parser_result<code> parse_expr(const uint8_t* pos)
+{
+    code c;
+    instr i;
+    do
+    {
+        i = static_cast<instr>(*pos);
+        switch (i)
+        {
+        default:
+            throw parser_error{"invalid instruction " + std::to_string(*pos)};
+
+        case instr::unreachable:
+        case instr::nop:
+        case instr::end:
+        case instr::local_get:
+        case instr::local_set:
+        case instr::local_tee:
+        case instr::i32_add:
+            break;
+        }
+        c.instructions.emplace_back(i);
+        ++pos;
```

Comment:
I think this should move into the case above. Each group of parsers will increment by the right amount.